src/lem/eval.rs

```diff
@@ -24,29 +24,31 @@ pub(crate) fn step() -> Result<LEM> {
 mod tests {
     use super::*;
     use crate::field::LurkField;
+    use crate::lem::constrainer::SlotsIndices;
     use crate::lem::{pointers::Ptr, store::Store};
     use bellperson::util_cs::{test_cs::TestConstraintSystem, Comparable};
     use blstrs::Scalar as Fr;
 
     const NUM_INPUTS: usize = 13;
-    const NUM_AUX: usize = 20;
+    const NUM_AUX: usize = 22;
     const NUM_CONSTRAINTS: usize = 29;
 
     fn test_eval_and_constrain_aux(store: &mut Store<Fr>, pairs: Vec<(Ptr<Fr>, Ptr<Fr>)>) {
         let lem = step().unwrap();
         for (expr_in, expr_out) in pairs {
-            let witnesses = lem.eval(expr_in, store).unwrap();
+            let valuations = lem.eval(expr_in, store).unwrap();
```

Comment:
`valuations` — it seems like a whole new lexicon is being created around this work. That's fine, I guess, but I bet there is an existing term that would more closely match whatever this is supposed to mean. As it is, the immediate context gives me little clue apart from what I can deduce about what this actually is. Do we need a new named concept for this?

Reply:
A "valuation" would be the result of the interpretation of a LEM. That is, the pairs of bindings of meta pointers to their respective real pointers. We don't have that on the current evaluation algorithm because those values are stored on Rust variables instead, which die after the Rust functions are finished.

I was calling it "witness" but @gabriel-barrett and @emmorais said that this is not the real witness, but only a subset of it, at best.

Reply:
Other possible words: `hints`, `non-deterministic-advices`, or just `advices`.